Modules/Sources/Yosemite/Stores/OrderCardPresentPaymentEligibilityStore.swift

```diff
@@ -55,11 +60,27 @@ private extension OrderCardPresentPaymentEligibilityStore {
                                               cardPresentPaymentsConfiguration: CardPresentPaymentsConfiguration,
                                               onCompletion: (Result<Bool, Error>) -> Void) {
         let storage = storageManager.viewStorage
+        let storageSite = storage.loadSite(siteID: siteID)?.toReadOnly()
+
+        let site: Site?
+        if let storageSite {
+            site = storageSite
+        } else {
+            /// Non - fatal fallback to `currentSite` when a storage site is missing
+            site = currentSite()
+
+            logFailedStorageSiteRead(
+                siteID: siteID,
+                currentSiteFallbackValue: site
+            )
+        }
```

Comment:
WDYT about checking currentSite() first and ensure that the current site has the same ID as siteID? I expect that it does, and we would not need to check storage unless otherwise.


```suggestion
        let site: Site? = {
            if let site = currentSite(), site.siteID == siteID {
                return site
            } 
            if let storageSite = storage.loadSite(siteID: siteID)?.toReadOnly() {
                return storageSite
            } else {
                logFailedStorageSiteRead(
                    siteID: siteID,
                    currentSiteFallbackValue: site
                )
                return nil
            }
        }()  
```